plugins/modules/virt.py

```diff
@@ -539,7 +543,10 @@ def core(module):
                 # Provide flag as an integer of all desired bits, see 'ENTRY_UNDEFINE_FLAGS_MAP'.
                 # Integer 23 takes care of all cases (23 = 1 + 2 + 4 + 16).
                 flag = 0
-                if force is True:
+                if mode is not None:
+                    for m in mode:
+                        flag += ENTRY_UNDEFINE_FLAGS_MAP.get(m, 0)
```

Comment:
@bturmann both `nvram` and `keep_nvram` are conflicting options, aren't that? I'm not sure what happens if both are set, but maybe we should have a check for this?

Reply:
Very good catch, @csmart. Indeed, they are conflicting options. I omitted a check, because my tests confirmed that libvirt library already refuses it with following rather clear error message:

```
$ ansible -b -m community.libvirt.virt -a "name=guest99 command=undefine mode=nvram,keep_nvram" host99
An exception occurred during task execution. To see the full traceback, use -vvv.
The error was: libvirt.libvirtError: 
Requested operation is not valid: cannot both keep and delete nvram
host99 | FAILED! => {
    "changed": false,
    "msg": "Requested operation is not valid: cannot both keep and delete nvram"
}
```

Therefore, another check here seems unnecessary. What do you think?

Reply:
Nice! I guess given that we're making it more advanced (and safe by default), we could leave it there and let it fail if someone happens to use conflicting flags. We just need to make sure we pass the error up the user, I guess.

Although my other thought is around what happens if the upstream behaviour changes, or different distros do something different? So perhaps an alternative is to implement a check ourselves and fail (protecting against upstream/distro changes and making the behaviour consistent) or we just default to safe and print a warning message? The latter would mean that the whole Ansible task wouldn't fail at least and the worst thing we've done is leave behind nvram, etc, which we'd notify the user about.

I think either approach works. What do you think? @Andersson007 any thoughts on best approach?

Reply:
Yes, we should catch potential errors. I was already working on it and about to push a new commit.